lib/facility_factory.rb

```diff
@@ -0,0 +1,43 @@
+class FacilityFactory
+
+    def initialize
+    end
+
+    def create_facilities(facilities)
+        format_data(facilities)
+        facilities.map do |facility|
+            Facility.new(facility)
+        end
+    end
+
+    def format_data(facilities)
+        if facilities.first.has_key?(:dmv_office)
+            co_format(facilities)
+        elsif facilities.first.has_key?(:address1)
+            mo_format(facilities)
+        else
+            ny_format(facilities)
+        end
+    end
+
+    def co_format(facilities)
+        facilities.map do |facility|
+            facility[:name] = facility.delete :dmv_office
+            facility[:address] = facility.values_at(:address_li, :address__1, :city, :state, :zip).join(" ")
+        end
+    end
+
+    def ny_format(facilities)
+        facilities.map do |facility|
+            facility[:name] = facility.delete :office_name
```

Comment:
Whether you map the existing facility hash or create a new hash with details and the key names you'd like, it doesn't really matter. This implementation is great